src/EventProducerService.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) Sven Rosvall (sven@rosvall.ie)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+#pragma once
+
+#include "Service.h"
+
+namespace VLCB {
+
+class Configuration;
+
+class EventProducerService : public Service {
+public:
+  virtual void setController(Controller *cntrl) override;
+  virtual void process(byte num); 
+  virtual Processed handleMessage(unsigned int opc, CANFrame *msg) override;
+
+  virtual byte getServiceID() override 
+  {
+    return 5;
+  }
+  virtual byte getServiceVersionID() override 
+  {
+    return 1;
+  }
+  void begin();
+  void sendEvent(bool state, byte index);
+  void sendEvent(bool state, byte index, byte data1);
+  void sendEvent(bool state, byte index, byte data1, byte data2);
+  void sendEvent(bool state, byte index, byte data1, byte data2, byte data3);
+
+private:
+  Controller *controller;
+  Configuration *module_config;  // Shortcut to reduce indirection code.
+  void (*eventhandler)(byte index, CANFrame *msg);
+  void (*eventhandlerex)(byte index, CANFrame *msg, bool evOn, byte evVal);
+
+  byte instantMode;
```

Comment:
It is confusing that you have an "instantMode" here and in MinimumNodeService. Do they do the same thing or just similar?

Reply:
They are similar in that they both note the mode at that instant to allow the service to determine where it is going.  Could change the instance in here to presentMode, if you like.

Reply:
No. Just confusing to have two similar names. So keep it as "instantMode" if it is needed.